apisix/plugins/ai-aws-content-moderation.lua

```diff
@@ -14,8 +14,12 @@
 -- See the License for the specific language governing permissions and
 -- limitations under the License.
 --
+require("resty.aws.config") -- to read env vars before initing aws module
```

Comment:
Do you want the default value of `false` for `AWS_EC2_METADATA_DISABLED` to cause access to IDMS to be turned off by default unless the user explicitly turns it on?

Can you give a more detailed argument for its use? 🤔

Reply:
I added this line mainly as a standard practice. (To read AWS specific ENV vars when APISIX is deployed in AWS environment)

> Do you want the default value of false for AWS_EC2_METADATA_DISABLED to cause access to IDMS to be turned off by default unless the user explicitly turns it on?

No, I added this to ensure the IDMS query can be disabled if need be.

Reply:
Therefore, it obeys those predefined environment variables anyway, and the initialization process for aws instance is moved to when the AWS API request actually occurs.
IDMS requests don't happen if the plugin configuration contains user-set AK/SK?

Reply:
> IDMS requests don't happen if the plugin configuration contains user-set AK/SK?

as per my last test, IDMS request happened anyway.

Reply:
So this will block requests when API calls are incoming, or block the secret manager.
Unless the user explicitly specifies `AWS_EC2_METADATA_DISABLED` as `true` to disable IDMS request?

Reply:
> So this will block requests when API calls are incoming, or block the secret manager.

yes, but only for the first time.